eng/pipelines/coreclr/templates/crossgen2-comparison-build-job.yml

```diff
@@ -0,0 +1,169 @@
+      - ${{ format('libraries_build_{0}{1}_{2}_{3}', parameters.osGroup, parameters.osSubgroup, parameters.archType, parameters.liveLibrariesBuildConfig) }}
+
+    # Run all steps in the container.
+    # Note that the containers are defined in platform-matrix.yml
+    container: ${{ parameters.container }}
+    timeoutInMinutes: 180 # 3 hrs
+
+    steps:
+
+    # Download product build
+    - template: /eng/pipelines/common/download-artifact-step.yml
+      parameters:
+        unpackFolder: $(buildProductRootFolderPath)
+        artifactFileName: '$(buildProductArtifactName)$(archiveExtension)'
+        artifactName: '$(buildProductArtifactName)'
+        displayName: 'product build'
+
+    # Optionally download live-built libraries
+    - ${{ if ne(parameters.liveLibrariesBuildConfig, '') }}:
+      - template: /eng/pipelines/common/download-artifact-step.yml
+        parameters:
+          unpackFolder: $(librariesDownloadDir)
+          cleanUnpackFolder: false
+          artifactFileName: '$(librariesBuildArtifactName)$(archiveExtension)'
+          artifactName: '$(librariesBuildArtifactName)'
+          displayName: 'live-built libraries'
+
+    # Populate Core_Root
+    - script: $(Build.SourcesDirectory)/src/tests/build$(scriptExt) $(buildConfig) $(archType) $(crossArg) generatelayoutonly
+      displayName: Populate Core_Root
+
+    # Create work item directory and populate with assemblies
+    - ${{ if ne(parameters.osGroup, 'Windows_NT') }}:
+      - script: |
+          mkdir -p $(workItemDirectory)
+          mkdir -p $(workItemDirectory)/log
+          mkdir -p $(workItemDirectory)/dlls
+          cp $(librariesProductDllDir)/* $(workItemDirectory)/dlls
+          cp $(productDirectory)/$(targetFlavor)/IL/System.Private.CoreLib.dll $(workItemDirectory)/dlls
+        displayName: Create directories
+    - ${{ if eq(parameters.osGroup, 'Windows_NT') }}:
+      - script: |
+          md $(workItemDirectory)\log
+          md $(workItemDirectory)\dlls
+          echo copy $(librariesProductDllDir)\* $(workItemDirectory)\dlls
+          copy $(librariesProductDllDir)\* $(workItemDirectory)\dlls
+          echo copy $(productDirectory)\$(targetFlavor)\IL\System.Private.CoreLib.dll $(workItemDirectory)\dlls
+          copy $(productDirectory)\$(targetFlavor)\IL\System.Private.CoreLib.dll $(workItemDirectory)\dlls
+        displayName: Create directories
+
+    # Create baseline output on the host (x64) machine
+    - task: PythonScript@0
+      displayName: Create cross-platform crossgen baseline
+      inputs:
+        scriptSource: 'filePath'
+        scriptPath: $(Build.SourcesDirectory)/src/tests/Common/scripts/crossgen2_comparison.py
+        ${{ if ne(parameters.osGroup, 'Windows_NT') }}:
+          pythonInterpreter: /usr/bin/python3
+          arguments:
+            crossgen_framework
+            --crossgen    $(crossgen2location)
+            --dotnet      $(Build.SourcesDirectory)/dotnet.sh
```

Comment:
Nit - at the first glance this Windows vs. Unix bifurcation seems solvable by using a few $(dir) substitutions (expanding to "/" on Unix and to "\" on Windows); if there's more to it let's leave this for a subsequent cleanup step.

Reply:
There is also the difference between running dotnet.sh and dotnet.cmd.